### ISSUE TITLE
Load lualdap module into openresty

### ISSUE THREAD

**Issue by tchimih** [openresty/lua-nginx-module#1267]
Hi,

I just want to know wether or how to use lualdap in openresty. I tried to download lua-ldap module but when I required it, it displayed:

`2018/02/21 09:49:41 [error] 27998#27998: *1 lua entry thread aborted: runtime error: /home/ubuntu/work/lua/test.lua:3: module 'lualdap' not found:
	no field package.preload['lualdap']
	no file '/usr/local/openresty/site/lualib/lualdap.ljbc'
	no file '/usr/local/openresty/site/lualib/lualdap/init.ljbc'
	no file '/usr/local/openresty/lualib/lualdap.ljbc'
	no file '/usr/local/openresty/lualib/lualdap/init.ljbc'
	no file '/usr/local/openresty/site/lualib/lualdap.lua'
	no file '/usr/local/openresty/site/lualib/lualdap/init.lua'
	no file '/usr/local/openresty/lualib/lualdap.lua'
	no file '/usr/local/openresty/lualib/lualdap/init.lua'
	no file './lualdap.lua'
	no file '/usr/local/openresty/luajit/share/luajit-2.1.0-beta3/lualdap.lua'
	no file '/usr/local/share/lua/5.1/lualdap.lua'
	no file '/usr/local/share/lua/5.1/lualdap/init.lua'
	no file '/usr/local/openresty/luajit/share/lua/5.1/lualdap.lua'
	no file '/usr/local/openresty/luajit/share/lua/5.1/lualdap/init.lua'
	no file '/usr/local/openresty/site/lualib/lualdap.so'
	no file '/usr/local/openresty/lualib/lualdap.so'
	no file './lualdap.so'
	no file '/usr/local/lib/lua/5.1/lualdap.so'
	no file '/usr/local/openresty/luajit/lib/lua/5.1/lualdap.so'
	no file '/usr/local/lib/lua/5.1/loadall.so'`

**Comment by tokers** [comment]
@tchimih What's your configuration about `lua_package_path`? You should move this module to a proper path where can be found according to the `lua_package_path`.

**Comment by tchimih** [comment]
Yes, I did it by copiying the lib inside /usr/local/openresty.  

Thanks for your quick reply :)

**Comment by tchimih** [comment]
Though, is it normal that my ldap search returns nothing, regarding the fact that the authentification went successful ... here is my code snippet:

` -- bind the user and check wether his credentials are truei
   ldap = assert(lualdap.open_simple ("192.168.0.54", "cn=admin,dc=my,dc=local","pwd"), usetls)

   if not ldap then
       return
   end

   ngx.log(ngx.ERR,"Attempt to search")
   for dn, attribs in ldap:search { base = "ou=People,dc=my,dc=local" } do
        ngx.log(ngx.ERR, string.format ("\t[%s]\n", dn))
   end`

**Comment by agentzh** [comment]
@tchimih Some points: 

1. Your lualdap library might be blocking the nginx event loop tragically upon network I/O if it's not utilizing OpenResty's cosocket API.
2. You should really contact the developers of that Lua library for support. If you believe it is this nginx module's issue, please provide a minimal and standalone example that can easily run our side and demonstrate the bug in a definite way. We have no time to dive into the implementation details of lualdap and debug everything in that 3rd-party Lua library for you.

You might have a better luck discussing such things in the openresty mailing list instead:

http://openresty.org/en/community.html

This place is for real bug report and development discussions in this nginx C module itself. Thanks for your cooperation.
